### PR TITLE
refact: Replace httparty in favor of core net/http package

### PR DESCRIPTION
**Pull request by oakbani**
## Summary
A request was made to minimize library dependencies from the SDK. HTTParty was being used at a minimal level only for fetching datafile and dispatching events. This PR replaces HTTParty by core ruby net/http package

## Test plan
All tests continue to pass. 

## Issues
https://github.com/optimizely/ruby-sdk/issues/211
OASIS-6453

### CODE REVIEW

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 1]
```diff
@@ -148,17 +150,16 @@ def request_config
         "Fetching datafile from #{@datafile_url}"
       )
       begin
-        headers = {
-          'Content-Type' => 'application/json'
-        }
+        uri = URI.parse(@datafile_url)
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.read_timeout = Helpers::Constants::CONFIG_MANAGER['REQUEST_TIMEOUT']
+        http.use_ssl = true
```
Comment (msohailhussain) [inline lib/optimizely/config_manager/http_project_config_manager.rb:156]: this is not true at all, datafile server can be without `ssl` if using custom URL.

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 2]
```diff
@@ -148,17 +150,16 @@ def request_config
         "Fetching datafile from #{@datafile_url}"
       )
       begin
-        headers = {
-          'Content-Type' => 'application/json'
-        }
+        uri = URI.parse(@datafile_url)
```
Comment (thomaszurkan-optimizely) [inline lib/optimizely/config_manager/http_project_config_manager.rb:153]: let's pull this out to a util method.

File: lib/optimizely/event_dispatcher.rb [hunk 3]
```diff
@@ -39,19 +39,29 @@ def initialize(logger: nil, error_handler: nil)
     #
     # @param event - Event object
     def dispatch_event(event)
+      uri = URI.parse(event.url)
```
Comment (thomaszurkan-optimizely) [inline lib/optimizely/event_dispatcher.rb:42]: util method?

File: lib/optimizely/event_dispatcher.rb [hunk 4]
```diff
@@ -39,19 +39,29 @@ def initialize(logger: nil, error_handler: nil)
...
-        response = HTTParty.post(event.url,
-                                 body: event.params.to_json,
-                                 headers: event.headers,
-                                 timeout: REQUEST_TIMEOUT)
+        request = Net::HTTP::Post.new(uri.request_uri)
+        request.body = event.params.to_json
+      end
+
+      # set headers
+      event.headers&.each do |key, val|
```
Comment (msohailhussain) [inline lib/optimizely/event_dispatcher.rb:56]: I don't see headers in old implementation.

Comment (oakbani) [reply lib/optimizely/event_dispatcher.rb:56]: We did pass headers to HTTPatty.post, which takes care of setting them on it's own.

File: lib/optimizely/config_manager/http_project_config_manager.rb (hunk 1 above)
Comment (oakbani) [reply lib/optimizely/config_manager/http_project_config_manager.rb:156]: nice catch

File: lib/optimizely/config_manager/http_project_config_manager.rb (hunk 2 above)
Comment (oakbani) [reply lib/optimizely/config_manager/http_project_config_manager.rb]: Done.

File: lib/optimizely/helpers/network_utils.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'net/http'
+
+module Optimizely
+  module Helpers
+    module NetworkUtils
```
Comment (msohailhussain) [inline lib/optimizely/helpers/network_utils.rb:23]: @thomaszurkan-optimizely Please suggest a name. 
`NetworkUtils` is quite broader term. `HttpUtils` would be more appropriate here.

**Pull request merged**